nailgun/entities.py

```diff
@@ -1876,6 +1876,7 @@ def __init__(self, server_config=None, **kwargs):
             'types': entity_fields.ListField(),
             'version': entity_fields.StringField(),
             'uuid': entity_fields.StringField(),
+            'architecture': entity_fields.OneToOneField(Architecture),
```

Comment:
I am facing issue with this, 
```
In [56]: arch[0]
Out[56]: nailgun.entities.Architecture(id=1, name=u'x86_64')
In [57]: entities.ContentViewFilterRule(server,content_view_filter=acvf,architecture=arch[0],name='test_arch1',version='2').create()
Out[57]: nailgun.entities.ContentViewFilterRule(version=u'2', name=u'test_arch1', id=7, content_view_filter=nailgun.entities.AbstractContentViewFilter(repository=[], inclusion=False, content_view=nailgun.entities.ContentView(id=20), description=None, type=u'rpm', id=8, name=u'qnplZzsAtK'))
```
Production log:
```
2018-08-03T09:42:07 [I|app|] Started POST "/katello/api/v2/content_view_filters/8/rules" for 10.67.116.203 at 2018-08-03 09:42:07 -0400
2018-08-03T09:42:07 [I|app|a3ad2] Processing by Katello::Api::V2::ContentViewFilterRulesController#create as */*
2018-08-03T09:42:07 [I|app|a3ad2]   Parameters: {"version"=>"2", "architecture_id"=>1, "name"=>"test_arch1", "content_view_filter_id"=>"8", "api_version"=>"v2", "content_view_filter_rule"=>{"version"=>"2", "architecture_id"=>1, "name"=>"test_arch1", "content_view_filter_id"=>"8"}}
2018-08-03T09:42:07 [I|app|a3ad2] Current user: foreman_admin (administrator)
2018-08-03T09:42:07 [I|app|a3ad2] Authorized user admin(Admin User)
2018-08-03T09:42:07 [I|app|a3ad2] Current user: admin (administrator)
2018-08-03T09:42:07 [I|app|a3ad2]   Rendering /opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.7/app/views/katello/api/v2/common/create.json.rabl within katello/api/v2/layouts/resource
2018-08-03T09:42:07 [I|app|a3ad2]   Rendered /opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.7/app/views/katello/api/v2/common/create.json.rabl within katello/api/v2/layouts/resource (3.4ms)
2018-08-03T09:42:07 [I|app|a3ad2] Completed 200 OK in 50ms (Views: 4.4ms | ActiveRecord: 12.0ms)
2018-08-03T09:42:32 [I|app|] Started GET "/katello/api/v2/content_view_filters/8/rules" for 10.67.116.203 at 2018-08-03 09:42:32 -0400
2018-08-03T09:42:32 [I|app|01e85] Processing by Katello::Api::V2::ContentViewFilterRulesController#index as */*
2018-08-03T09:42:32 [I|app|01e85]   Parameters: {"content_view_filter_id"=>"8", "api_version"=>"v2", "content_view_filter_rule"=>{"content_view_filter_id"=>"8"}}
2018-08-03T09:42:33 [I|app|01e85] Current user: foreman_admin (administrator)
2018-08-03T09:42:33 [I|app|01e85] Authorized user admin(Admin User)
2018-08-03T09:42:33 [I|app|01e85] Current user: admin (administrator)
2018-08-03T09:42:33 [I|app|01e85]   Rendering /opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.7/app/views/katello/api/v2/content_view_filter_rules/index.json.rabl within katello/api/v2/layouts/collection
2018-08-03T09:42:33 [I|app|01e85]   Rendered /opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.7/app/views/katello/api/v2/content_view_filter_rules/index.json.rabl within katello/api/v2/layouts/collection (16.6ms)
```
Filter is not able to associate the arch, I also checked from UI.
Looks like even though controller doesn't complain about taking in  "architecture_id"=>1 , it still needs architecture as String.
I guess this would be a product issue?